.github/workflows/test-publish-r2.yml

```diff
@@ -0,0 +1,75 @@
+---
+name: Test publish-r2 flow
+
+on:
+  workflow_dispatch:
+  push:
+    # Only trigger when pushing to the 'main' branch
+    # AND only if this specific workflow file changes.
+    branches:
```

Comment:
I think this should be invoked on PRs too? Only triggering on main wouldn't prevent regressions from getting merged...